api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/StringHandler.java

```diff
@@ -263,6 +277,48 @@ public void addStringData(
         observable.subscribe(new ResultSetObserver(asyncResponse));
     }
 
+    @POST
+    @Path("/raw/query")
+    @ApiOperation(value = "Fetch raw data points for multiple metrics")
+    public Response findRawData(QueryRequest query) {
+        TimeRange timeRange = new TimeRange(query.getStart(), query.getEnd());
+        if (!timeRange.isValid()) {
+            return badRequest(new ApiError(timeRange.getProblem()));
+        }
+
+        int limit;
+        if (query.getLimit() == null) {
+            limit = 0;
+        } else {
+            limit = query.getLimit();
+        }
+        Order order;
+        if (query.getOrder() == null) {
+            order = Order.defaultValue(limit, timeRange.getStart(), timeRange.getEnd());
+        } else {
+            order = Order.fromText(query.getOrder());
+        }
+
+        List<MetricId<String>> metricIds = query.getIds().stream().map(id -> new MetricId<>(tenantId, STRING, id))
+                .collect(toList());
+        Observable<NamedDataPoint<String>> dataPoints = metricsService.findDataPoints(metricIds, timeRange.getStart(),
+                timeRange.getEnd(), limit, order).observeOn(Schedulers.io());
+
+        StreamingOutput stream = output -> {
+            JsonGenerator generator = mapper.getFactory().createGenerator(output, JsonEncoding.UTF8);
+            CountDownLatch latch = new CountDownLatch(1);
+            dataPoints.subscribe(new NamedDataPointObserver<>(generator, latch, STRING));
+
+            try {
+                latch.await();
```

Comment:
I disagree that it goes against our reactive architecture because we do not have a fully async/non-blocking architecture. Undertow may be async, but JAX-RS certainly is not. In every endpoint in which we use AsyncResponse we block that thread until the response is fully realized in memory. It simply is not fully reactive. With these new endpoints, we can be dealing with arbitrarily large result sets that could easily introduce lots of heap pressure, making the server not so scalable. I think that avoiding that memory pressure is more important than deferring the work to another thread which will still block until the response is fully realized.

I tried to make the streaming work with AsyncResponse but was unsuccessful. If there is a way to make it work, then I am all for it. If we were running on a fully async stack, then what I have done with the CountDownLatch would be a big red flag.